### PR TITLE
RDK-35853: Crash observed when few plugins are activated from Control…

### PR DESCRIPTION
**Pull request by kathiravancomcast**
…ler UI in CMF-RPi

From: kchinn681 <kathiravan_chinnadurai@comcast.com>

Subject: Crash observed when few plugins are activated from Controller UI in CMF-RPi
Reason for change:Crash observed when few plugins are activated from Controller UI in CMF-RPi
Test Procedure: activate html app using controller.ui
Risks: Low
Signed-off-by: kathiravan chinnadurai <kathiravan_chinnadurai@comcast.com>
Source: COMCAST
License: GPLV2
Upstream-Status: Pending

### CODE REVIEW

File: RDKShell/RDKShell.cpp [hunk 1]
```diff
@@ -544,6 +545,24 @@ namespace WPEFramework {
             }
             return exist;
         }
+        static void setSurfaceClientName( void)
```
Comment (madanagopalt) [inline RDKShell/RDKShell.cpp:548]: rename function as updateSurfaceClientIdentifiers

File: RDKShell/RDKShell.cpp [hunk 2]
```diff
@@ -544,6 +545,24 @@ namespace WPEFramework {
...
+        {
+	  std::vector <string>::iterator it;
+	  for(it=gSurfaceClient.begin();it!=gSurfaceClient.end();it++)
+          {
+           uint32_t status = 0;
+           auto thunderController = getThunderControllerClient();
```
Comment (madanagopalt) [inline RDKShell/RDKShell.cpp:554]: move this line outside loop

File: RDKShell/RDKShell.cpp [hunk 3]
```diff
@@ -1119,7 +1138,11 @@ namespace WPEFramework {
             m_timer.setInterval(RECONNECTION_TIME_IN_MILLISECONDS);
             m_timer.start();
             std::cout << "Started SystemServices connection timer" << std::endl;
-
+            char* rdkshelltype = getenv("RDKSHELL_COMPOSITOR_TYPE");
+            if (strcmp(rdkshelltype , "surface") == 0)
```
Comment (madanagopalt) [inline RDKShell/RDKShell.cpp:1142]: have a null check for rdkshelltype as below:
if ((NULL != rdkshelltype ) && (strcmp(rdkshelltype , "surface") == 0))

**Comment by Firdose2109** [comment]
Can one of the admins verify this patch?

**Comment by anand-ky** [comment]
Closing older PRs. Pls resubmit to latest 2205 sprint.

**Pull request closed without merging**